### DISCUSSION TITLE
JS Based Export not working when using ['react-app'] in .baberc

### DISCUSSION THREAD

**Discussion by ashatrisk** [parcel-bundler/parcel#6327] [Q&A] [answered]
I am creating multi page react application using parcel. I have created components folder & it has components which i can import. In all the react component i am using 
`var React = require('react')`
& it is working fine for react components. But in JavaScript export it is also asking for react. Since it throws error in console
_Uncaught ReferenceError: React is not defined_  

How to export JS based import in parcel react environment?

Thanks!!

my package.json
`"devDependencies":{
	"@babel/core" : "^7.13.10",
	"@babel/plugin-proposal-class-properties" : "^7.13.10",
	"babel-preset-env": "^1.7.0",
	"babel-preset-react-app": "^10.0.0",
	"parcel-bundler": "^1.12.5"
}
"dependencies":{
	"@popperjs/core": "^2.9.2",
	"bootstrap": "^5.0.1",
	"node-sass": "^5.0.0",
	"react": "^17.0.2",
	"react-dom": "^17.0.2",
	"react-router-dom": "^5.2.0"
}`

here is menubarjs.js

```
function menubarjs(){
	var str= 
				<div id="navbar" class="collapse navbar-collapse">
                            <ul class="nav navbar-nav">
                                <li>
                                    <a href="./Home">Home <span class="btm-brdr"></span></a>
                                </li>
								<li class="active">
                                    <a href="#">Gallery
								  <span class="btm-brdr"></span>
								  </a>
                                </li>
								 <li>
                                    <a href="./service">Service  <span class="btm-brdr"></span></a>
                                </li>
                                <li class="contact">
                                    <a href="./contact">Contact US  <span class="btm-brdr"></span></a>
                                </li>
                               
                                
                            </ul>

                            <span class="dashboard"> Dashboard </span>
                        </div>
	
	return str
}

export default menubarjs
```

**Comment by mischnic** [comment]
Please provide a complete code sample and the Parcel version you're using.

**Comment by ashatrisk** [comment, marked as answer]
I found out what been wrong there, I missed quoting str value. After quoting in "", it worked fine. I wasted lot of time for that simple stuff
